python/pyarrow/config.pxi

```diff
@@ -74,3 +74,21 @@ def runtime_info():
     return RuntimeInfo(
         simd_level=frombytes(c_info.simd_level),
         detected_simd_level=frombytes(c_info.detected_simd_level))
+
+
+def configure_tzdb(path):
```

Comment:
I might go for a more explicit name, using "timezone" in full. Given the name of the option on the C++ side, maybe `set_timezone_db_path`?